lib/matobj.gi

```diff
@@ -277,6 +288,8 @@ InstallMethod( ZeroVector,
 #M  Matrix( <list>, <ncols> )
 #M  Matrix( <list> )
 ##
+##  Compute the missing arguments for 'NewMatrix' and then call it.
+##
 InstallMethod( Matrix,
   [ IsOperation, IsSemiring, IsList, IsInt ],
   { filt, R, list, nrCols } -> NewMatrix( filt, R, nrCols, list ) );
```

Comment:
Wasn't the idea to insert (optional) checks into these `Matrix` functions, to validate things like that `nrCols` and `Length(list)` are compatible, that the given lists don't have holes, that entries are contained in the given basedomain etc ?

Reply:
My point of view is as follows:

The documentation says that implementing new types of matrix objects requires the installation of a `NewMatrix` method,
whereas there are default `Matrix` methods which compute the missing arguments for `NewMatrix`.
The question whether the arguments of `Matrix` (and hence the arguments that are then passed to `NewMatrix`) are consistent depends on the type of the matrix objects in question; it may be required that all matrix entries lie in the base domain, or more general situations are possible. The documentation is (deliberately?) vague in this respect, and adding the concrete consistency conditions to the `NewMatrix` method in question will make precise what is allowed for the given type of matrix objects.

I think that consistency checks in `Matrix` methods can be either only partial checks (which need more checks in the `NewMatrix` methods) or default checks (which need individual `Matrix` methods with modified checks if the default does not fit). Wouldn't that be more complicated in the end?